File: server/models/Customer.js
class Customer {
  constructor(id, name, email, contactNumber, address) {
    this.id = id;
    this.name = name;
    this.email = email;
    this.contactNumber = contactNumber;
    this.address = address;
    this.appointments = [];
    this.feedbacks = [];
  }

  bookAppointment(appointment) {
    this.appointments.push(appointment);
  }

  cancelAppointment(appointmentId) {
    this.appointments = this.appointments.filter(appt => appt.id !== appointmentId);
  }

  rescheduleAppointment(appointmentId, newAppointment) {
    this.cancelAppointment(appointmentId);
    this.bookAppointment(newAppointment);
  }

  giveFeedback(appointmentId, message, rating) {
    const feedback = { appointmentId, message, rating };
    this.feedbacks.push(feedback);
  }
}

module.exports = Customer;

File: server/app/models/Customer.java
package app.models;

import java.util.ArrayList;
import java.util.List;

public class Customer {
    private String id;
    private String name;
    private String email;
    private String contactNumber;
    private List<Appointment> appointments;
    private List<Feedback> feedbacks;

    public Customer(String id, String name, String email, String contactNumber) {
        this.id = id;
        this.name = name;
        this.email = email;
        this.contactNumber = contactNumber;
        this.appointments = new ArrayList<>();
        this.feedbacks = new ArrayList<>();
    }

    // Getters and Setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getContactNumber() { return contactNumber; }
    public void setContactNumber(String contactNumber) { this.contactNumber = contactNumber; }

    public List<Appointment> getAppointments() { return appointments; }
    public void setAppointments(List<Appointment> appointments) { this.appointments = appointments; }

    public List<Feedback> getFeedbacks() { return feedbacks; }
    public void setFeedbacks(List<Feedback> feedbacks) { this.feedbacks = feedbacks; }

    // Business Methods
    public void addAppointment(Appointment appointment) {
        appointments.add(appointment);
    }

    public void removeAppointment(Appointment appointment) {
        appointments.remove(appointment);
    }

    public void addFeedback(Feedback feedback) {
        feedbacks.add(feedback);
    }

    @Override
    public String toString() {
        return "Customer{" +
                "id='" + id + '\'' +
                ", name='" + name + '\'' +
                ", email='" + email + '\'' +
                ", contactNumber='" + contactNumber + '\'' +
                '}';
    }
} 